js/src/setup-ads/ads-stepper/index.js

```diff
@@ -27,12 +28,18 @@ import {
  */
 const AdsStepper = ( { formProps } ) => {
 	const [ step, setStep ] = useState( '1' );
+	const { hasFinishedResolution, hasGoogleAdsConnection } =
+		useGoogleAdsAccount();
 
 	useEventPropertiesFilter( FILTER_ONBOARDING, {
 		context: CONTEXT_ADS_ONBOARDING,
 		step,
 	} );
 
+	if ( ! hasFinishedResolution && ! hasGoogleAdsConnection ) {
```

Comment:
If we use `glaData.initialWpData.adsId`, there won't be the need to show a spinner since the data is already available.

Reply:
Initially I was concerned about value not being available when we manually reload the page, as this was the case when I was using `hasGoogleAdsConnection`. But with use of `glaData.initialWpData.adsId` worked totally fine.